README.md

```diff
@@ -550,6 +550,32 @@ console.log(newName); // ['Ryan', 'McDermott'];
 ```
 **[⬆ back to top](#table-of-contents)**
 
+### Avoid Side Effects pt.2
+Side effects could also occur from inside a function. In Javascript, function arguments 
+are always passed by value except when they(functions) are passed reference values such as 
+objects and arrays. In that case, we should be carefull not to change any of these 
+argument's properties. A possible solution would be to always clone the variable,
+edit it and return the clone.
+
+**Bad:**
+```javascript
+function userReducer(state, action) {
+  state.userProfile = action.payload;
+  return state;
+}
+```
+
+**Good:**
+```javascript
```

Comment:
This is a good start. I think we can do something that's not Flux/Redux specific, as these examples can go over the heads of a lot of beginners. Simple things like Planes, Trains, and Automobiles are understood by everyone! Let's think of an example along those lines.